src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/INodeFilter.cs

```diff
@@ -53,5 +53,25 @@ public static string GetPropertyFilter(IEnumerable<INodeFilter> filters, Type ty
             }
             return string.Empty;
         }
+
+        /// <summary>
+        /// Appends a list of property filters to an URN query string
+        /// </summary>
+        public static string AddPropertyFilterToFilterString(string filterString, IEnumerable<INodeFilter> filters, Type type, ValidForFlag validForFlag)
+        {
+            if(String.IsNullOrEmpty(filterString))
+            {
+                return GetPropertyFilter(filters, type, validForFlag);
+            }
+            foreach(var value in filters)
+            {
+                var filter = value.ToPropertyFilterString(type, validForFlag);
+                if(filter != string.Empty)
+                {
+                    filterString = filterString.Remove(filterString.Length - 1, 1) + $" and {filter}" + "]";
```

Comment:
What's the last square bracket for?

You could also get rid of the extra append operation by just adding the square bracket to the end of the interpolated string.

`filterString = filterString.Remove(filterString.Length - 1, 1) + $" and {filter}]"

Reply:
That's the last format for URN https://learn.microsoft.com/en-us/sql/powershell/query-expressions-and-uniform-resource-names?view=sql-server-ver16